FindmallocMC.cmake

```diff
@@ -158,5 +178,29 @@ find_package_handle_standard_args(mallocMC
     REQUIRED_VARS ${mallocMC_REQUIRED_VARS_LIST}
     VERSION_VAR mallocMC_VERSION
     HANDLE_COMPONENTS
-    )
+)
+
+# clean up
+#
+# unset cached variables in case we did not find a valid install
+# (e.g., we only found an outdated version)
+if(NOT mallocMC_FOUND)
+    # default vars
+    unset(mallocMC_VERSION)
+    foreach(REQ_VAR ${mallocMC_REQUIRED_VARS_LIST})
+        unset(${REQ_VAR})
+        unset(${REQ_VAR} CACHE)
+    endforeach()
+
+    # user-level component vars
+    foreach(COMPONENT ${mallocMC_FIND_COMPONENTS})
+        if(mallocMC_${COMPONENT}_FOUND)
```

Comment:
yes, see [the code where it checks if the halloc include_dir was found](https://github.com/ComputationalRadiationPhysics/cmake-modules/blob/topic-mallocPreferHints/FindmallocMC.cmake#L152-L156) for an example of this actually happening.